docs/wave.rst

```diff
@@ -0,0 +1,106 @@
+  data.set_receiver_pos(receiver_coords)
+  data.set_shape(nt, 101)
+
+First of all, we will set up the initial condition for damping::
+
+  from devito.interfaces import DenseData
+  self.damp = DenseData(name="damp", shape=self.model.get_shape_comp(),
+                        dtype=self.dtype)
+  # Initialize damp by calling the function that can precompute damping
+  damp_boundary(self.damp.data)
+
+DenseData is a devito data object used to store and manage spatially varying data.
+
+:samp:`damp_boundary()` function initialises the damp on each grid point. The initial data are stored in :obj:`self.damp.data`.
+
+Similarly, we will set up the initial condition and allocate the grid for m.
+Initial value of m on each grid point is stored as a numpy array in :obj:`m.data[:]`.::
+
+  m = DenseData(name="m", shape=model.get_shape_comp(), dtype=damp.dtype)
+  m.data[:] = model.padm()
+
+after that, we will initilisize u::
+
+  u = TimeData(name="u", shape=model.get_shape_comp(), time_dim=nt,
+                   time_order=time_order, space_order=spc_order, save=True,
+                   dtype=damp.dtype)
+
+TimeData is a devito data object used to store and manage time-varying data
+
+We initialise our grid to be of size :samp:`model.get_shape_comp()` which is a 3-D tuple.
+:obj:`time_dim` represents the size of the time dimension that dictates the leading dimension of the data buffer.
+:obj:`time_order` and :obj:`space_order` represent the discretization order for time and space respectively.
+
+The next step is to generate the stencil to be solved by a :obj:`devito.operator.Operator`
+
+The stencil is generated according to Devito conventions. It uses a sympy
+expression to represent the acoustic wave equation. Devito makes it easy to
+represent the equation in a finite-difference form by providing properties :obj:`dt2`, :obj:`laplace`, :obj:`dt`.
+We then generate the stencil by solving eqn for u.forward, a symbol for the time-forward state of the function.
+::
+
+  from devito import Operator
+  from sympy import Eq, solve, symbols
+  eqn = m*u.dt2-u.laplace+damp*u.dt
+  stencil = solve(eqn, u.forward)[0]
+
+We plug the stencil in an Operator, as shown, and define the values the spacing between cells :obj:`h` and the
+temporal spacing :obj:`s`.::
+
+  s, h = symbols('s h')
+  subs = {s: model.get_critical_dt(), h: model.get_spacing()}
+  super(AdjointOperator, self).__init__(nt, m.shape,
+                                        stencils=Eq(v.backward, stencil),
+                                        substitutions=subs,
+                                        spc_border=spc_order/2,
+                                        time_order=time_order,
+                                        forward=False,
+                                        dtype=m.dtype,
+                                        **kwargs)
+
+
+To execute the generated Operator, we simply call :samp:`op.apply()`. The results will then be found in :obj:`u.data`
+
```

Comment:
`self.propagator.time_loop_stencils_a = src.add(m, u) + rec.read(u)`
At least this line, for clarity.